src/rust/engine/src/nodes.rs

```diff
@@ -487,6 +487,61 @@ impl From<Scandir> for NodeKey {
   }
 }
 
+///
+/// A node that captures Vec<PathStat> for resolved files/dirs from PathGlobs.
+///
+/// This is similar to the Snapshot node, but avoids writing to LMDB store as a performance
+/// optimization.
+///
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+pub struct FileListing(pub Key);
+
+impl FileListing {
+  async fn create(context: Context, path_globs: PreparedPathGlobs) -> NodeResult<Vec<PathStat>> {
+    context
+      .expand_globs(path_globs)
+      .map_err(|e| throw(&format!("{}", e)))
+      .await
+  }
+
+  pub fn store_file_listing(core: &Arc<Core>, item: &[PathStat]) -> Result<Value, String> {
+    let mut files = Vec::new();
+    let mut dirs = Vec::new();
+    for ps in item.iter() {
+      match ps {
+        &PathStat::File { ref path, .. } => {
+          files.push(Snapshot::store_path(path)?);
+        }
+        &PathStat::Dir { ref path, .. } => {
+          dirs.push(Snapshot::store_path(path)?);
+        }
+      }
+    }
+    Ok(externs::unsafe_call(
+      core.types.file_listing,
+      &[externs::store_tuple(files), externs::store_tuple(dirs)],
+    ))
+  }
+}
+
+#[async_trait]
+impl WrappedNode for FileListing {
+  type Item = Vec<PathStat>;
```

Comment:
The output type of `Nodes` will be `clone()`d fairly often. You might see whether you can measure the difference between this being an `Arc<Vec<PathStat>>` and not, and if you can't, leaving it as is.

Reply:
Ah, that's why you were using an `Arc` for `Snapshot`. Whereas `hashing::Digest` implements `Copy`, so it doesn't need a wrapper.

I'll make the change, as you were using `Arc` before for snapshot.